### PR TITLE
Support fast scrolling using a modifier key #2363

### PR DESCRIPTION
**Pull request by csurquin**
Here is my attempt to implement this [feature](https://github.com/xtermjs/xterm.js/issues/2363). The sensitivity is adjustable and the modifier key can be chosen from 'alt', 'ctrl', 'shift' or undefined to disable the feature.

I did not add automated tests because I was not sure how to do it but I may work on it later following your feedback.

### CODE REVIEW

File: demo/client.ts [hunk 1]
```diff
@@ -228,6 +228,7 @@ function initOptions(term: TerminalType): void {
     bellSound: null,
     bellStyle: ['none', 'sound'],
     cursorStyle: ['block', 'underline', 'bar'],
+    fastScrollModifier: ['alt', 'ctrl', 'shift', undefined],
```
Comment (Tyriar) [inline demo/client.ts]: Not sure undefined will work here? This drives the dropdown in the demo

Comment (csurquin) [reply demo/client.ts]: I work on the development container and it seems to behave as expected. Here is what is displayed in the demo.

demo
If I select undefined, then none of the modifier keys activates the fast scrolling.

**Comment by csurquin** [comment]
I will try to implement a test as soon as possible. Since it is my first pull request I have a question, I read that it is possible to update a pull request by pushing additional commits on the branch you submitted. However, I naively pushed my commits on `master`. Will it work the same way?

File: demo/client.ts (hunk 1 above)
Comment (Tyriar) [reply demo/client.ts]: Perfect 👌

**Comment by Tyriar** [comment]
@csurquin you can just push to master, you just won't be able to do another PR without extra hoops. I'd suggest finishing this one off first, then in the future checkout a branch and submit any more against that (`git checkout -b myfeature`)

**Pull request closed without merging**